### PR TITLE
Add pointwise sum

### PR DESCRIPTION
**Pull request by adler-j**
Added PointwiseSum utility by using the PointwiseInner functionality.

In the future, we can optimize this.

### CODE REVIEW

File: odl/discr/tensor_ops.py [hunk 1]
```diff
@@ -669,6 +669,67 @@ def adjoint(self):
...
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        vfspace : `ProductSpace`
+            Space of vector fields on which the operator acts.
+            It has to be a product space of identical spaces, i.e. a
+            power space.
+        weight : `array-like` or float, optional
+            Weighting array or constant for the norm. If an array is
```
Comment (adler-j) [inline odl/discr/tensor_ops.py]: norm -> sum. General problem.

File: odl/discr/tensor_ops.py [hunk 2]
```diff
@@ -669,6 +669,67 @@ def adjoint(self):
...
+# TODO: Optimize this to an optimized operator on its own.
+class PointwiseSum(PointwiseInner):
+
+    """Take the point-wise sum of a vector field.
+
+    This operator takes the (weighted) sum
+
+        ``sum(F(x)) = [ sum_j( w_j * F_j(x) ) ]
+
+    where ``F`` is a vector field. This implies that
```
Comment (kohr-h) [inline odl/discr/tensor_ops.py]: No objections against "vector field"? Here it wouldn't be a big pain to change to "vector-valued function" which is a more generous term. The variable name could even stay :-)

File: odl/discr/tensor_ops.py [hunk 3]
```diff
@@ -669,6 +669,67 @@ def adjoint(self):
...
+    def __init__(self, vfspace, weight=None):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        vfspace : `ProductSpace`
+            Space of vector fields on which the operator acts.
+            It has to be a product space of identical spaces, i.e. a
+            power space.
+        weight : `array-like` or float, optional
```
Comment (kohr-h) [inline odl/discr/tensor_ops.py]: positive float
Then remove the corresponding sentence from the text below.

**Comment by adler-j** [comment]
> Fix the comment you had yourself and that one I had. There's also the more general question about "vector field" against "vector-valued function". Cool otherwise.

Well this whole file is positioned wrong, since it actually works on any linear space and has nothing to do with discretizations, frankly it would play better inside productspace operators or something like that. Will fix comments.

**Comment by kohr-h** [comment]
This will become even more interesting when we have tensors. Because then, all these operators are just standard operators that only act along the first axis.

File: odl/discr/tensor_ops.py (hunk 2 above)
Comment (adler-j) [inline odl/discr/tensor_ops.py]: Vector valued function implies it is actually a function in a stronger sense than vector field does, at least to me. This function accepts anything, i.e. a productspace of rn spaces is fine. So we dont want a very strong notion of function in there.

File: odl/discr/tensor_ops.py (hunk 3 above)
Comment (adler-j) [inline odl/discr/tensor_ops.py]: I went by simply removing the positivity constraint. It is not really a strict constraint anyway, everything works with negative values.

**Comment by adler-j** [comment]
Updated by comments, but changed to remove the positivity condition that was rather arbitrary anyway. Needs new review I guess.

**Comment by kohr-h** [comment]
> Updated by comments, but changed to remove the positivity condition that was rather arbitrary anyway. Needs new review I guess.

With "unbiased" operators like sum or inner product the constraint really was arbitrary and can be removed, but for anything that has the name "norm" in it I'm against allowing negative weigths. That will catch users off guard.

**Comment by adler-j** [comment]
> With "unbiased" operators like sum or inner product the constraint really was arbitrary and can be removed, but for anything that has the name "norm" in it I'm against allowing negative weigths. That will catch users off guard.

Agree and that was my plan, will see I went over board.

**Comment by adler-j** [comment]
> Agree and that was my plan, will see I went over top.

It seems I actually succeeded. Any more comments?

File: odl/test/discr/tensor_ops_test.py [hunk 4]
```diff
@@ -89,12 +89,6 @@ def test_pointwise_norm_init_properties():
     with pytest.raises(ValueError):
         PointwiseNorm(vfspace, exponent=0.5)  # < 1 not allowed
 
-    with pytest.raises(ValueError):
-        PointwiseNorm(vfspace, weight=-1)  # < 0 not allowed
-
-    with pytest.raises(ValueError):
-        PointwiseNorm(vfspace, weight=[1, 0, 1])  # 0 invalid
```
Comment (kohr-h) [inline odl/test/discr/tensor_ops_test.py]: Add these back

**Comment by kohr-h** [comment]
Otherwise LGTM.

**Comment by adler-j** [comment]
Merge after CI

**Pull request merged**